config/config.proto

```diff
@@ -48,9 +48,24 @@ message TargetGroup {
 	optional LabelPairs labels = 2;
 }
 
+// Information about the service discovery.
+message ServiceDiscovery {
+	// The name by which the service information can be found for
+	// the specified discovery service.
+	required string name = 1;
+	// The type of discovery to be used.
+	required string type = 2;
+	// Path to a configuration file that is used for setting up
```

Comment:
I think we should use the main proto file for any required configuration, such as via proto extensions.